techmatters/hrm
# chore: Create reindex cases admin endpoint [CHI-2759]

## Description
Analogous to https://github.com/techmatters/hrm/pull/651, this PR introduces an admin endpoint, `admin-url.../cases/reindex` that will trigger a reindex of the cases for the specified `accountSid`, which are within the range of `dateFrom` and `dateTo`.
This endpoint is also exposed via the admin-cli script introduced in https://github.com/techmatters/hrm/pull/652.

### Checklist
- [x] [Corresponding issue has been opened](https://tech-matters.atlassian.net/browse/CHI-2759)

### Verification steps
- Make sure this PR is deployed to HRM development.
- Connect to the devops instance.
- From within it, run 
  `$ npm run admin-cli reindex hrm -- -a=ACd8a2e89748318adf6ddff7df6948deaf -e=development -r=us-east-1 -f=2024-01-01 -t=2024-03-01 --ca
`
- Once above finishes:
  - Check the [hrm logs in couldwatc](https://us-east-1.console.aws.amazon.com/cloudwatch/home?region=us-east-1#logsV2:log-groups/log-group/development-hrm-awslogs) and confirm the `publishToSearchIndex` debug logs are visible (use "filter pattern" feature here).
  - Go to ES cloud console and confirm that the cases are properly indexed.


### AFTER YOU MERGE

1. Cut a release tag using the GitHub workflow. Wait for it to complete and notify in the #aselo-deploys Slack channel.
2. Comment on the ticket with the release tag version AND any additional instructions required to configure an environment to test the changes.
3. Only then move the ticket into the QA column in JIRA

You are responsible for ensuring the above steps are completed. If you move a ticket into QA without advising what version to test, the QA team will assume the latest tag has the changes. If it does not, the following confusion is on you! :-P
